src/powerquery-language-services/providers/languageCompletionItemProvider.ts

```diff
@@ -26,17 +25,19 @@ export class LanguageAutocompleteItemProvider implements AutocompleteItemProvide
         KeywordKind.HashTime,
     ];
 
-    constructor(private readonly maybeTriedInspection: WorkspaceCache.InspectionCacheItem) {}
+    constructor(private readonly maybePromiseInspection: Promise<Inspection.Inspection | undefined>) {}
```

Comment:
When we pass around a `Promise` like this, it becomes shared, correct? meaning, the first await will ensure the promise completes, and the second returns the previous result?

Reply:
So long as they're the same reference, then yes. That's why I still have the WorkspaceCache to begin with, I needed somewhere to store shared references. The current strategy is to void the cahce if a previously seen uri has a different version number than the cached analysis.